File: collecting_images.r
# install.packages('dplyr')
# install.packages('RCurl')
# install.packages('rvest')

library(dplyr)
library(RCurl)
library(rvest)

# 宣言
dir.default <- getwd()

url.top <- 'http://example.net/'
url.search <- 'example.net/dummy'
url.replace.1 <- 'example.net/dummy/img/'
url.replace.2 <- 'example.net/src/img/'
url.posts <- paste('page-', 1:1000000, sep='')

i <- 0

# page
sapply(url.posts, function(url.post) {
  i <<- i + 1
  cat(i, '\t / \t', length(url.posts), '\t: ', url.post, '\t: ')

  tryCatch({
    cat('\t')
    html.post <- read_html(paste(url.top, url.post, sep=''), encoding='UTF-8')
    html.post.title <- html.post %>% html_nodes(xpath='//title') %>% html_text()
    write(paste(url.post, '\t', url.post, '\t', html.post.title, '\t\t', sep=''), 'list.txt', append = T)
    html.post.links <- html.post %>% html_nodes(xpath='//div/a[@target="_blank"]') %>% html_attr('href')
    html.post.links <- html.post.links[grep(url.search, html.post.links)]
    url.post.imgs <- sub(url.replace.1, url.replace.2, html.post.links)
    df <- data.frame(FILE=basename(url.post.imgs), IMG=url.post.imgs, LINK=html.post.links)

    if(nrow(df)>0) {
      cat(length(url.post.imgs), '\t: ')

      dir.create(sub('[?]', '', url.post))
      setwd(paste('./', sub('[?]', '', url.post), sep=''))

      # an image
      apply(df, 1, function(img) {
        tryCatch({
          cat('.')
          writeBin(getBinaryURL(img['IMG'], referer = img['LINK']), img['FILE'])
        },
        warning = {},
        error = {},
        silent = TRUE)
      })
    }

    html.post <- NULL
  },
  warning = function(w) {cat('w')},
  error = function(e) {cat('e')},
  finally = {
    setwd(dir.default)

    cat('\n')
  },
  silent = TRUE)
})

File: collecting_images.py
#-*- coding:utf-8 -*-
# Copyright (c) 2017 YA-androidapp(https://github.com/YA-androidapp) All rights reserved.

# pip install beautifulsoup4
# pip install chardet

import bs4
import chardet
import os
import sys
import urllib.request

# 宣言
dir_default = os.getcwd()

url_top = 'http://example.net/'
url_search = 'example.net/dummy'
url_replace_1 = 'example.net/dummy/img/'
url_replace_2 = 'example.net/src/img/'

max = 1000000

# page
i = 0
for i in range(0, max, 1):
    i = i + 1
    url_post = 'page-' + str(i)
    print(str(i) + '\t / \t' + str(max) +
          '\t: ' + url_post + '\t: ', end='')

    try:
        print('\t', end='')

        request = urllib.request.urlopen(url_top + url_post)
        html_post = request.read()
        guess = chardet.detect(html_post)
        html_post = html_post.decode(guess['encoding'])
        soup = bs4.BeautifulSoup(html_post)

        html_post_title = soup.title.string

        try:
            file_list_txt = open('list.txt', 'a')
            file_list_txt.write(url_post + '\t' + url_post +
                                '\t' + html_post_title + '\t\t')
            file_list_txt.close()
        except:
            pass

        html_post_links = soup.find_all('a', target='_blank')

        if(len(html_post_links) > 0):
            print(str(len(html_post_links)) + '\t: ', end='')

            os.mkdir('./' + url_post.replace('?', ''))
            os.chdir('./' + url_post.replace('?', ''))

            # an image
            for img_link in html_post_links:
                try:
                    if img_link.get('href').find(url_search) != -1:
                        print('.', end='')
                        img_img = img_link.get('href').replace(
                            url_replace_1, url_replace_2)
                        req = urllib.request.Request(img_img)
                        headers = {'Referer': img_link.get('href')}
                        req = urllib.request.Request(
                            img_img, headers=headers)
                        with urllib.request.urlopen(req) as response:
                            f = open(os.path.basename(img_img), 'wb')
                            f.write(response.read())
                except:
                    print(sys.exc_info())
    except:
        print(sys.exc_info())
    finally:
        os.chdir(dir_default)
